spec/services/school_groups/categorise_schools_spec.rb

```diff
@@ -0,0 +1,253 @@
+  let(:school_10) { create(:school, id: 10, name: "School 10", school_group: school_group, active: true, visible: true) }
+  let(:school_11) { create(:school, id: 11, name: "School 11", school_group: school_group, active: true, visible: true) }
+  let(:school_12) { create(:school, id: 12, name: "School 12", school_group: school_group, active: true, visible: true) }
+  let(:school_13) { create(:school, id: 13, name: "School 13", school_group: school_group, active: true, visible: true) }
+  let(:school_14) { create(:school, id: 14, name: "School 14", school_group: school_group, active: true, visible: true) }
+  let(:school_15) { create(:school, id: 15, name: "School 15", school_group: school_group, active: true, visible: true) }
+  let(:school_16) { create(:school, id: 16, name: "School 16", school_group: school_group, active: true, visible: true) }
+  let(:school_17) { create(:school, id: 17, name: "School 17", school_group: school_group, active: true, visible: true) }
+  let(:school_18) { create(:school, id: 18, name: "School 18", school_group: school_group, active: true, visible: true) }
+  let(:school_19) { create(:school, id: 19, name: "School 19", school_group: school_group, active: true, visible: true) }
+  let(:school_20) { create(:school, id: 20, name: "School 20", school_group: school_group, active: true, visible: true) }
+  let(:school_21) { create(:school, id: 21, name: "School 21", school_group: school_group, active: true, visible: true) }
+  let(:school_22) { create(:school, id: 22, name: "School 22", school_group: school_group, active: true, visible: true) }
+  let(:school_23) { create(:school, id: 23, name: "School 23", school_group: school_group, active: true, visible: true) }
+  let(:school_24) { create(:school, id: 24, name: "School 24", school_group: school_group, active: true, visible: true) }
+  let(:school_25) { create(:school, id: 25, name: "School 25", school_group: school_group, active: false, visible: true) }
+  let(:school_26) { create(:school, id: 26, name: "School 26", school_group: school_group, active: true, visible: false) }
+
+  let(:school_27) { create(:school, id: 27, name: "School 27", school_group: school_group_2, active: true, visible: true) }
+  let(:school_28) { create(:school, id: 28, name: "School 28", school_group: school_group_2, active: true, visible: true) }
+  let(:school_29) { create(:school, id: 29, name: "School 29", school_group: school_group_2, active: true, visible: true) }
+  let(:school_30) { create(:school, id: 30, name: "School 30", school_group: school_group_2, active: true, visible: true) }
+
+  let!(:baseload_advice_pages) { create(:advice_page, key: 'baseload') }
+  let!(:electricity_intraday_advice_pages) { create(:advice_page, key: 'electricity_intraday') }
+  let!(:electricity_long_term_advice_pages) { create(:advice_page, key: 'electricity_long_term') }
+
+  before do
```

Comment:
Is there scope to DRY up the spec by making more use of FactoryBot and maybe some traits or helpers to avoid long lists of data? e.g, something like. 

```
create_list(:school, 10, :with_advice_page_benchmarks, advice_page_benchmarks: { base_load_advice_page: :exemplar, electricity_intraday_page: :well_managed})
```

It'd make the specs easier to read.

Reply:
I've moved the upsert to a method so it's a little easier to read now.  I think for now i'll leave as is and make a note/ticket to revisit the spec when the group pages are finalised.